crates/edr_solidity_tests/src/gas_report.rs

```diff
@@ -115,17 +115,36 @@ impl GasReport {
             contract_info.size = trace.data.len();
         } else if let Some(DecodedCallData { signature, .. }) = decoded().await.call_data {
             let name = signature.split('(').next().unwrap();
-            // ignore any test/setup functions
+            // Contract deployment status is determined by the setUp function
+            let is_setup = name.test_function_kind().is_setup();
+            // Ignore any test functions
             let should_include = !name.test_function_kind().is_known();
-            if should_include {
+            if is_setup {
+                contract_info.status = if trace.is_error() {
+                    GasReportExecutionStatus::Halt
+                } else if trace.is_revert() {
+                    GasReportExecutionStatus::Revert
+                } else {
+                    GasReportExecutionStatus::Success
+                };
```

Comment:
Polish: Can the code duplication between these lines and lines 138-146 be easily avoided?